pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -508,6 +518,7 @@ func (oe *operationExecutor) generateDetachVolumeFunc(
 	}
 
 	return func() error {
+		ref, errGetRef := api.GetReference(oe.PodRef.Pod)
```

Comment:
ok, if there is an error the implication is that the event will not be generated, so I will create a log entry to that effect